omnipaxos/src/sequence_paxos/follower.rs

```diff
@@ -114,7 +204,11 @@ where
             });
             match accsync.stopsign {
                 Some(ss) => {
-                    if let Some(ss_entry) = self.internal_storage.get_stopsign() {
+                    if let Some(ss_entry) = self
```

Comment:
Yes, I think we should always explicitly send `AcceptStopSign` and `DecideStopSign`. So we should remove the stopsign field from the `Prepare` message and then modify the leader code so that we instead explicitly send `AcceptStopSign` and possibly `DecideStopSign` if we adopt a StopSign in the prepare phase (in `handle_promise()`).

Reply:
After looking into it a little more, I think the current behavior is almost correct.
During the prepare phase, stopsigns are only accepted and followers do send `AcceptedStopsign` if they receive one in the `AccSync`. The only thing is that they might never find out that the stopsign is decided.
I think the easiest fix should be to respond with `DecideStopsign` in `handle_accepted_stopsign` if the stopsign was already decided, instead of just ignoring the msg.

Reply:
And looking into it even more, even that problem seems non-existent to me, since the leader seems to broadcast `DecideStopsign` in that case.
As a small optimization I'd only make the leader write to his storage if his stopsign was previously undecided, but overall it seems to me, that everything should actually work.

And I previously forgot, but of course the possible problem of implicitly deciding the stopsign in `AccSync` seems to still exist.

On that topic: why do we need 2 sources of truth in storage on whether a stopsign is decided? Shouldn't a fix be to remove that redundancy?

Reply:
In general the dropped `DecideStopSign` is a bit tricky. I think we will solve it by periodically resending the decide to all promised servers periodically. I think @kevin-harrison will fix that in PR soon to complete #68.

Reply:
It is because we wanted the stored log to be of type `Entry`. This makes it easier to read/write from storage and implement snapshots (i.e., snapshots can be created from an iterator of `Entry`) for the user.

Reply:
> On that topic: why do we need 2 sources of truth in storage on whether a stopsign is decided? Shouldn't a fix be to remove that redundancy?

It is because we wanted the stored log to be of type `Entry`. This makes it easier to read/write from storage and implement snapshots (i.e., snapshots can be created from an iterator of `Entry`) for the user.

Reply:
But at the same time, we wanted the stopsign to appear as the last log entry in the log e.g., when reading, therefore the decided index also might include the StopSign

Reply:
Okay but then why does the `decided_idx` need to also reflect that the stopsign is decided? Can't we just say `decided_idx` is only for entries and the stopsign is only decided by its own flag?
So we would then not increment `decided_idx` on deciding a stopsign.

Reply:
Ah I didn't see your last comment before writing that

Reply:
This will be fixed in a separate PR, since it's not related to this change.